src/main/java/duke/TextDuke.java

```diff
@@ -31,15 +33,15 @@ private void run() {
      * Sets up the required objects, loads the data from the storage file, and displays the welcome message.
      */
     private void start() {
-        ui = new Ui();
+        textUi = new TextUi();
         parser = new Parser();
         try {
             taskList = new TaskList(Storage.load());
         } catch (IOException e) {
-            ui.showLoadingError(e.getMessage());
+            textUi.showLoadingError(e.getMessage());
             taskList = new TaskList();
         }
-        ui.showWelcome();
+        textUi.showWelcome();
```

Comment:
Minor point, but maybe just leave this as "welcome" instead of "showWelcome"?

Reply:
I wanted to have `show` in the method name to indicate an action, will change to `showWelcomeMessage` to be more specific.